llvm/lib/Transforms/InstCombine/InstCombinePHI.cpp

```diff
@@ -697,8 +697,7 @@ static bool isSafeAndProfitableToSinkLoad(LoadInst *L) {
 Instruction *InstCombinerImpl::foldPHIArgLoadIntoPHI(PHINode &PN) {
   LoadInst *FirstLI = cast<LoadInst>(PN.getIncomingValue(0));
 
-  // Can't forward swifterror through a phi.
-  if (FirstLI->getOperand(0)->isSwiftError())
+  if (!shouldFoldOperandThroughPhi(FirstLI->getOperand(0)))
```

Comment:
I think my general preference would be to not add a new API for this and just call canReplaceOperandWithVariable() here.